lib/nm/index.ts

```diff
@@ -0,0 +1,377 @@
+import * as Bluebird from 'bluebird';
+import * as _ from 'lodash';
+import {NetworkManagerTypes} from './types';
+
+const dbus: any = require('dbus-native');
+
+const systemBus: any = dbus.systemBus();
+const service: string[] = [ 'org', 'freedesktop', 'NetworkManager' ];
+
+export interface Device {
+	path: string;
+	DeviceType: number;
+}
+
+export interface AccessPoint {
+	path: string;
+	Ssid: string;
+}
+
+export class NetworkManager extends NetworkManagerTypes {
+	accessPoints: any[];
+	wifiActive: boolean;
+	devices: {
+		wifi: Device;
+	};
+
+	constructor() {
+		super();
+
+		this.devices = {
+			wifi: {
+				path: '',
+				DeviceType: 0,
+			},
+		};
+	}
+
+	init() {
+		return this.getWifiDevice();
+	}
+
+	async toggleWifi(value) {
+		if (value) {
+			const [err, success] = await to(this.connectDevice(this.devices.wifi.path));
+			if (err) {
+				throw formatError(500)(`Could not toggleWifi`)(err);
```

Comment:
Also, why aren't we exposing the original error?

Reply:
What do you mean with `expose the original error`? It's the last parameter of `formatError` and it is included in the `return`ed object of that function.
I agree on converting the return value to a classic `Error`, the function is meant to be just a simple wrapper to centralize error management so that all functions follow the same error-handling schema (which means it'll be easier to swap everything to the Error class).

Reply:
Also, wouldn't swapping to the `Error` class let the use of only the error `message` as a string, locking out the error `code` and the `err` object itself?